src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,45 +3,165 @@
 import java.util.List;
 
 public class MyLinkedList<T> implements MyLinkedListInterface<T> {
+    private Node<T> tail;
+    private int size;
+    private Node<T> head;
+
     @Override
     public void add(T value) {
```

Comment:
When creating a new Node, you should pass all the fields to the Node constructor. You are currently doing this correctly, but please ensure it is consistent throughout your code.